ciao-controller/configuration.go

```diff
@@ -0,0 +1,265 @@
+// Copyright (c) 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/01org/ciao/ciao-controller/types"
+	"github.com/01org/ciao/payloads"
+	"github.com/01org/ciao/service"
+	"github.com/01org/ciao/ssntp"
+	"github.com/golang/glog"
+	"gopkg.in/yaml.v2"
+)
+
+func (c *controller) UpdateClusterConfig(ctx context.Context, newConf types.ConfigRequest, token string) error {
+	var pyld payloads.Configure
+	tenantID, err := service.GetTenantID(ctx)
```

Comment:
No, the TenantID is get here because later on is used in order to test a request with an URL that contains a tenant.

For administrative privileges, the validation is done according to the route specified [here](https://github.com/01org/ciao/pull/1130/files#diff-42c37cb0629f871056ad7039b938c272R769) by telling that the specified route requires to be done by privileged tenant:
```
route = r.Handle("/configuration", Handler{context, showConfig, true})
                                                                ^^^^
```